src/Datadog.Trace.DuckTyping/DuckType.cs

```diff
@@ -0,0 +1,509 @@
+                {
+                    DuckTypePropertyOrFieldNotFoundException.Throw(proxyProperty.Name, duckAttribute.Name);
+                }
+            }
+        }
+
+        private static CreateProxyInstance GetCreateProxyInstanceDelegate(Type proxyType, Type targetType)
+        {
+            ConstructorInfo ctor = proxyType.GetConstructors()[0];
+
+            DynamicMethod converterMethod = new DynamicMethod(
+                $"CreateProxyInstance<{proxyType.Name}>",
+                typeof(IDuckType),
+                new[] { typeof(object) },
+                typeof(DuckType).Module,
+                true);
+            ILGenerator il = converterMethod.GetILGenerator();
+            il.Emit(OpCodes.Ldarg_0);
+            if (targetType.IsPublic || targetType.IsNestedPublic)
+            {
+                il.Emit(OpCodes.Castclass, targetType);
+            }
+
+            il.Emit(OpCodes.Newobj, ctor);
+
+            if (proxyType.IsValueType)
+            {
+                il.Emit(OpCodes.Box, proxyType);
+            }
+
+            il.Emit(OpCodes.Ret);
+            return (CreateProxyInstance)converterMethod.CreateDelegate(typeof(CreateProxyInstance));
+        }
+
+        /// <summary>
+        /// Struct to store the result of creating a proxy type
+        /// </summary>
+        public readonly struct CreateTypeResult
+        {
+            /// <summary>
+            /// Gets if the proxy type creation was successful
+            /// </summary>
+            public readonly bool Success;
+
+            /// <summary>
+            /// Target type
+            /// </summary>
+            public readonly Type TargetType;
+
+            private readonly Type _proxyType;
+            private readonly ExceptionDispatchInfo _exceptionInfo;
+            private readonly CreateProxyInstance _activator;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="CreateTypeResult"/> struct.
+            /// </summary>
+            /// <param name="proxyType">Proxy type</param>
+            /// <param name="targetType">Target type</param>
+            /// <param name="activator">Proxy activator</param>
+            /// <param name="exceptionInfo">Exception dispatch info instance</param>
+            internal CreateTypeResult(Type proxyType, Type targetType, CreateProxyInstance activator, ExceptionDispatchInfo exceptionInfo)
+            {
+                _proxyType = proxyType;
+                TargetType = targetType;
+                _activator = activator;
+                _exceptionInfo = exceptionInfo;
+                Success = proxyType != null && exceptionInfo == null;
+            }
+
+            /// <summary>
+            /// Gets the created ProxyType
+            /// </summary>
+            public Type ProxyType
+            {
+                get
+                {
+                    _exceptionInfo?.Throw();
+                    return _proxyType;
+                }
+            }
```

Comment:
Unused property. Removing it would allow us to remove one field from the struct (which is always a good thing :p )

Reply:
Yep I know, I have it because I'm not 100% sure if I'll will use this property for CallTarget invoker or not, I think I'll remove it for now, and add it later in case is needed.

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/946/commits/56f255754ee4a37b0893ef8c40db42d8f44fc390